Cannot read field "parent" because "comp" is null

com.intellij.diagnostic.PluginException: Cannot read field "parent" because "comp" is null [Plugin: com.redhat.devtools.lsp4ij]
	at com.intellij.ui.EditorNotificationsImpl$updateEditors$job$1.invokeSuspend(EditorNotificationsImpl.kt:267)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:102)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
Caused by: java.lang.NullPointerException: Cannot read field "parent" because "comp" is null
	at java.desktop/java.awt.Container.remove(Container.java:1290)
	at com.intellij.openapi.fileEditor.impl.EditorComposite.manageTopOrBottomComponent(EditorComposite.kt:542)
	at com.intellij.openapi.fileEditor.impl.EditorComposite.removeTopComponent(EditorComposite.kt:526)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.removeTopComponent(FileEditorManagerImpl.kt:1737)
	at com.intellij.ui.EditorNotificationsImpl.updateNotification(EditorNotificationsImpl.kt:286)
	at com.intellij.ui.EditorNotificationsImpl.access$updateNotification(EditorNotificationsImpl.kt:57)
	at com.intellij.ui.EditorNotificationsImpl$updateEditors$job$1$2.invokeSuspend$lambda$0(EditorNotificationsImpl.kt:258)
	at com.intellij.openapi.application.CoroutinesKt.writeIntentReadAction$lambda$1$lambda$0(coroutines.kt:329)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:128)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:917)
	at com.intellij.openapi.application.CoroutinesKt.writeIntentReadAction$lambda$1(coroutines.kt:329)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContextInner(coroutines.kt:341)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invokeSuspend(coroutines.kt:233)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at com.intellij.openapi.progress.CoroutinesKt$blockingContext$2.invoke(coroutines.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:261)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:232)
	at com.intellij.openapi.application.CoroutinesKt.writeIntentReadAction(coroutines.kt:328)
	at com.intellij.ui.EditorNotificationsImpl$updateEditors$job$1$2.invokeSuspend(EditorNotificationsImpl.kt:253)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:221)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:117)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:728)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:750)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:675)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:573)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18$lambda$17$lambda$16$lambda$15(IdeEventQueue.kt:355)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:857)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18$lambda$17$lambda$16(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2$lambda$1(IdeEventQueue.kt:1045)
	at com.intellij.openapi.application.WriteIntentReadAction.lambda$run$0(WriteIntentReadAction.java:24)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.runWriteIntentReadAction(AnyThreadWriteThreadingSupport.kt:128)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteIntentReadAction(ApplicationImpl.java:917)
	at com.intellij.openapi.application.WriteIntentReadAction.compute(WriteIntentReadAction.java:55)
	at com.intellij.openapi.application.WriteIntentReadAction.run(WriteIntentReadAction.java:23)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$2(IdeEventQueue.kt:1045)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$3(IdeEventQueue.kt:1054)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:109)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1054)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$18(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:395)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

--- a/src/main/java/com/redhat/devtools/lsp4ij/ui/LanguageServerSuggestionEditorNotificationProvider.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/ui/LanguageServerSuggestionEditorNotificationProvider.java
@@ -11,6 +11,7 @@
 package com.redhat.devtools.lsp4ij.ui;
 
 import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.fileEditor.FileEditor;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
@@ -138,9 +139,11 @@ public class LanguageServerSuggestionEditorNotificationProvider implements Edito
                 NewLanguageServerDialog dialog = new NewLanguageServerDialog(project);
                 dialog.loadFromTemplate(template);
                 dialog.show();
-                // Refresh notifications to hide it
-                EditorNotifications.getInstance(project)
-                        .updateAllNotifications();
+                // Refresh notifications to hide it (deferred to avoid race condition)
+                ApplicationManager.getApplication().invokeLater(() -> {
+                    EditorNotifications.getInstance(project)
+                            .updateAllNotifications();
+                });
             });
         });
 
@@ -148,9 +151,11 @@ public class LanguageServerSuggestionEditorNotificationProvider implements Edito
         panel.createActionLabel(LanguageServerBundle.message("language.server.suggest.install.dismiss"), () -> {
             // Store preference to disable future notifications
             PropertiesComponent.getInstance().setValue(DISABLE_KEY, true);
-            // Refresh notifications to hide the panel
-            EditorNotifications.getInstance(project)
-                    .updateAllNotifications();
+            // Refresh notifications to hide the panel (deferred to avoid race condition)
+            ApplicationManager.getApplication().invokeLater(() -> {
+                EditorNotifications.getInstance(project)
+                        .updateAllNotifications();
+            });
         });
 
         return panel;

